### PR TITLE
Add validation of filepaths for non-BIDS NWB assets

### LINKED ISSUE #1163: upload doesn't seem to be checking for valid DANDI (when not BIDS) organization of files/assets
it seems that some aspects of dandiset naming/layout may be falling through the cracks. here is another example: https://dandiarchive.org/dandiset/000168/draft

as part of healthcheck if may be worthwhile considering if files should have a different name, but doesn't. this could capture a few things: 1) dandisets like this one, 2) dandisets that were uploaded in parts, and may need name resolution before upload.

### PR DESCRIPTION
**Pull request by jwodder**
Closes #1163.

### CODE REVIEW

File: dandi/organize.py [hunk 1]
```diff
@@ -1025,3 +1028,75 @@ def msg_(msg, n, cond=None):
         msg_(" %d invalid not considered.", skip_invalid),
         dandiset_path.rstrip("/"),
     )
+
+
+LABELREGEX = r"[^_*\\/<>:|\"'?%@;.]+"
+ORGANIZED_FILENAME_REGEX = (
+    rf"sub-{LABELREGEX}(_(ses|tis|slice|cell)-{LABELREGEX})*(_[a-z]+(\+[a-z]+)*)?\.nwb"
```
Comment (yarikoptic) [inline dandi/organize.py:1035]: - let's make at least `_ses` strictly following `sub-` if present
- according to https://github.com/dandi/dandi-cli/blob/HEAD/dandi/organize.py#L45 is still missing `probe` and `obj`.

Comment (yarikoptic) [reply dandi/organize.py:1035]: ideally there should be a single data structure describing entities etc we should have in the filename. In principle that `potential_fields` is the one but I guess can't be used here "as is", unless you enrich that structure with regexes for the values, in particular for extensions etc.  Then this regex  can be constructed programmatically and we would avoid duplication of this knowledge.

Comment (jwodder) [reply dandi/organize.py:1035]: @yarikoptic Can tis/slice/cell/probe/obj occur without a preceding ses, or is ses a prerequisite for them?

Comment (yarikoptic) [reply dandi/organize.py:1035]: according to `potential_fields`, `ses` is not mandatory, so other entities could occur without a preceding `ses`.

Comment (jwodder) [reply dandi/organize.py]: Done: 55df999.

**Comment by yarikoptic** [comment]
ok, I dislike that we have duplication of specification ATM between organize and validate but ok -- let's RF it after.

Meanwhlie filed #1174 which found while test driving, and seems to work:

```
❯ dandi validate XCaMPgf/XCaMPgf_ANM471996_cell01.nwb
[NWBI.check_subject_species_form] XCaMPgf/XCaMPgf_ANM471996_cell01.nwb — Subject species 'mus musculus' should be in latin binomial form, e.g. 'Mus musculus' and 'Homo sapiens'
[NWBI.check_intracellular_electrode_cell_id_exists] XCaMPgf/XCaMPgf_ANM471996_cell01.nwb — Please include a unique cell_id associated with this IntracellularElectrode.
[DANDI.NON_DANDI_FILENAME] XCaMPgf/XCaMPgf_ANM471996_cell01.nwb — Filename does not conform to Dandi standard
[DANDI.NON_DANDI_FOLDERNAME] XCaMPgf/XCaMPgf_ANM471996_cell01.nwb — File is not in folder at root with subject name
```

although messages are a bit cryptic I guess.

**Pull request merged**